src/Esfa.Vacancy.Register.Infrastructure/Repositories/VacancyRepository.cs

```diff
@@ -13,14 +14,26 @@ public class VacancyRepository : IVacancyRepository
     {
         private const string GetLiveVacancyByReferenceNumberSqlSproc = "[VACANCY_API].[GetLiveVacancy]";
         private readonly IProvideSettings _provideSettings;
+        private readonly ILog _logger;
 
-        public VacancyRepository(IProvideSettings provideSettings)
+        public VacancyRepository(IProvideSettings provideSettings, ILog logger)
         {
             _provideSettings = provideSettings;
+            _logger = logger;
         }
 
         public async Task<DomainEntities.Vacancy> GetVacancyByReferenceNumberAsync(int referenceNumber)
         {
+            var retry = VacancyRegisterRetryPolicy.GetFixedIntervalPolicy((exception, time, retryCount, context) =>
+            {
+                _logger.Warn($"Error retrieving vacancy from VacancyRepository: ({exception.Message}). Retrying...attempt {retryCount})");
```

Comment:
Could you say database instead of 'VacancyRepository' as the log message will already have a 'logger' field with VacancyRepository